### ISSUE TITLE
`no-inline-assertions` rule

### ISSUE THREAD

**Issue by sindresorhus** [avajs/eslint-plugin-ava#152]
[<img alt="Issuehunt badges" src="https://img.shields.io/badge/IssueHunt-%2460%20Funded-%2300A156.svg" />](https://issuehunt.io/r/avajs/eslint-plugin-ava/issues/152)

For readability reasons, I try to not use inline arrow functions when the function doesn't return, to make it clear it doesn't return. I would like to prevent assertions from being called in an inline arrow function test. This might seem like an edge-case, but I've seen it many times and have had to comment about it on multiple pull requests.
## Fail

``` js
test('foo', t => t.true(fn()));
```
## Pass

``` js
test('foo', t => {
    t.true(fn());
});
```

---

<details>
<summary>
<b>IssueHunt Summary</b>
</summary>

### Sponsors (Total: $60.00)

- [<img src='https://avatars3.githubusercontent.com/u/44827199?v=4' alt='issuehunt' width=24 height=24> issuehunt](https://issuehunt.io/u/issuehunt) ($60.00)

### Tips

- Checkout the [Issuehunt explorer](https://issuehunt.io/r/avajs/eslint-plugin-ava/) to discover more funded issues.
- Need some help from other developers? [Add your repositories](https://issuehunt.io/r/new) on IssueHunt to raise funds.
</details>

**Comment by jfmengels** [comment]
Sounds good to me.

**Comment by IssueHuntBot** [comment]
[@issuehunt](https://issuehunt.io/sponsors) has funded $60.00 to this issue.

---
- Submit pull request via [IssueHunt](https://issuehunt.io/repos/49722492/issues/152) to receive this reward.
- Want to contribute? Chip in to this issue via [IssueHunt](https://issuehunt.io/repos/49722492/issues/152).
- Checkout the [IssueHunt Issue Explorer](https://issuehunt.io/issues) to see more funded issues.
- Need help from developers? [Add your repository](https://issuehunt.io/r/new) on IssueHunt to raise funds.

**Comment by JLHwung** [comment]
I would like to work on it in couple of days.

**Comment by sindresorhus** [comment]
Go ahead :)

Make sure you carefully read https://github.com/avajs/eslint-plugin-ava/blob/master/docs/new-rule.md first and also look at some previous PRs to see how it's done.
